### PR TITLE
fix(har timing): record `connect` timing for proxied connections

### PR DESCRIPTION
**Pull request by Skn0tt**
Fixes a bug discovered in https://github.com/microsoft/playwright/pull/32647. When using http proxy, the `connect` event isn't emitted so we don't populate `tcpConnectionAt`. The updated version of `https-proxy-agent` emits a `proxyConnect` as a replacement, so this PR updates and listens to that event.
For socks proxies, the `on("socket")` event is emitted once the SOCKS connection is established, which is the equivalent of having a TCP connection available.

### CODE REVIEW

File: packages/playwright-core/src/server/fetch.ts [hunk 1]
```diff
@@ -680,17 +689,16 @@ export class GlobalAPIRequestContext extends APIRequestContext {
...
+    return new SocksProxyAgent(proxyURL);
+
   if (proxy.username)
-    proxyOpts.auth = `${proxy.username}:${proxy.password || ''}`;
+    proxyURL.username = proxy.username;
+  if (proxy.password)
+    proxyURL.password = proxy.password;
   // TODO: We should use HttpProxyAgent conditional on proxyOpts.protocol instead of always using CONNECT method.
-  return new HttpsProxyAgent(proxyOpts);
+  return new HttpsProxyAgent(proxy.server);
```
Comment (mxschmitt) [inline packages/playwright-core/src/server/fetch.ts:701]:
```suggestion
  return new HttpsProxyAgent(proxyURL);
```

shouldn't it be like that?

Comment (Skn0tt) [reply packages/playwright-core/src/server/fetch.ts:701]: yes! :facepalm:

Comment (Skn0tt) [reply packages/playwright-core/src/server/fetch.ts:701]: yes! :facepalm:

**Comment by dgozman** [comment]
Do we have to upgrade the deps? I am always wary of introducing regressions.

**Comment by Skn0tt** [comment]
For SOCKS proxy we don't, but for HTTP proxy yes. The `proxyConnect` event was only added in `6.1.0` and without it we have no way of knowing when the proxy started the connection.

If we don't want to update, another alternative would be leave things as for the HTTP proxy and set `connect: -1`.

**Comment by Skn0tt** [comment]
Could you give this another look @dgozman?

**Review by dgozman — APPROVED** [review]

**Pull request merged**